src/core/weights.js

```diff
@@ -48,6 +53,42 @@ export function copy(w: Weights): Weights {
   };
 }
 
+/**
+ * Merge multiple Weights together.
+ *
+ * The resultant Weights will have every weight specified by each of the
+ * input weights. If the inputs provide contradictory weights for any node
+ * or edge address, an error will be thrown.
+ */
+export function merge(ws: $ReadOnlyArray<Weights>): Weights {
+  const weights = empty();
```

Comment:
Nit: because `empty()` here doesn't mention weights. Would like if the variable was typed.
`const weights: Weights = empty();`

Reply:
Up to @decentralion, but this looks fine to me as is. We know that it’s
a `Weights` because the variable is called `weights`, the module is
`weights.js`, and the function takes a bunch of `Weights` and returns a
`Weights`. I don’t think that this suggestion makes the code materially
better.

Reply:
Yeah this is a nit, cognitive load preference.
> the module is
weights.js, and the function takes a bunch of Weights and returns a
Weights

Having the typed variable doesn't have me thinking "did we import `empty` from somewhere else?", "being named `weights`, I'm assuming it's `Weights` typed... right?".